src/robot/libraries/Collections.py

```diff
@@ -14,31 +14,41 @@
 #  limitations under the License.
 
 import copy
-from ast import literal_eval
+from collections.abc import Hashable
 from itertools import chain
+from typing import (
+    Any, Iterable, Iterator, Literal, Mapping, MutableMapping, MutableSequence,
+    NoReturn, overload, Sequence, Union
+)
 
 from robot.api import logger
 from robot.utils import (
-    is_dict_like, is_list_like, Matcher, NotSet, plural_or_not as s, seq2str, seq2str2,
-    type_name
+    is_list_like, Matcher, NotSet, plural_or_not as s, seq2str, seq2str2, type_name
 )
 from robot.utils.asserts import assert_equal
 from robot.version import get_version
 
 NOT_SET = NotSet()
 
+IgnoreKeyCase = Literal["KEY", "KEYS"]
+IgnoreValueCase = Literal["VALUE", "VALUES"]
```

Comment:
I realized we actually shouldn't have separate key and value literals but should always use a literal with both. The reason is that normalization is recursive and, for example, a list can have dicts that should be normalized. This probably should get tests.